modules/openapi-generator/src/main/resources/kotlin-ktor-client/infrastructure/HttpClientConvenience.kt.mustache

```diff
@@ -69,6 +70,37 @@ internal suspend inline fun <reified T, B> request(
     )
 }
 
+internal suspend inline fun <reified T, B> requestWithResponse(
```

Comment:
instead of having a separate function for this and returning a generic map,
i would prefer if we parse the headers as described in https://swagger.io/docs/specification/v3_0/describing-responses/#response-headers
and then return a type that has properties for the described headers (with correct type, as well)